core/util/src/main/java/org/eclipse/rdf4j/common/iteration/CloseableIteration.java

```diff
@@ -33,4 +35,15 @@
 	@Override
 	public void close() throws X;
 
+	/**
+	 *
+	 * Convert the results to a Java 8 Stream. Stream should be closed by calling .close().
```

Comment:
Calling `close()` is only necessary if the stream is not fully consumed: the `Spliterator` implementation we use automatically closes the iteration under the hood when exhausted (or when an exception occurs). Perhaps we can reword to something like:

> The stream should be closed (by calling `Stream#close()`) if it is not fully consumed. 

Arguably, that is something that should be done with every Stream object anyway, so this could just be added to the javadoc of `Iteration.stream`, removing the need for the override here.